contributors/design-proposals/daemonset-update.md

```diff
@@ -146,13 +169,25 @@ type DaemonSetStatus struct {
 	// (ready for at least minReadySeconds)
 	// +optional
 	NumberUnavailable int32 `json:"numberUnavailable"`
+
+	// A monotonically increasing counter that tracks hash collisions for
+	// the DaemonSet. Used as a collision avoidance mechanism by the 
+	// DaemonSet controller.
+	// +optional
+	Uniquifier *int64 `json:"uniquifier,omitempty"`
 }
 
 const (
+	// DEPRECATED: DefaultDeploymentUniqueLabelKey is used instead.
 	// DaemonSetTemplateGenerationKey is the key of the labels that is added
-	// to daemon set pods to distinguish between old and new pod templates
+	// to daemon set pods to distinguish between old and new pod
 	// during DaemonSet template update.
 	DaemonSetTemplateGenerationKey string = "pod-template-generation"
+
+	// DefaultDaemonSetUniqueLabelKey is the default label key that is added
+	// to existing DaemonSet pods to distinguish between old and new
+	// DaemonSet pods during DaemonSet template updates.
+	DefaultDaemonSetUniqueLabelKey string = "pod-template-hash"
```

Comment:
This is not final. #594 proposed hashing `AnyState.Data`. If we decided to hash data instead of template, this won't be a proper name. If all controllers (except for Deployments) will use `AnyStates`, we can just use `AnyState.Name` which is guaranteed to be unique.